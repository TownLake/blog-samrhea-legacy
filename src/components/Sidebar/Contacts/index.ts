export { default as Contacts } from "./Contacts";
