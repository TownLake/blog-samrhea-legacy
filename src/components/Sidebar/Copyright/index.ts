export { default as Copyright } from "./Copyright";
