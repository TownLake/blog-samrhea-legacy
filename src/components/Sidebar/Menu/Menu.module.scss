@import "../../../assets/scss/variables";
@import "../../../assets/scss/mixins";

.menu {
  @include margin-bottom(1);

  .list {
    list-style: none;
    @include margin-equal(0);
    @include padding-equal(0);

    .item {
      @include margin(0.5, 0);
      @include padding-equal(0);

      .link {
        border: 0;
        color: var(--color-typographic-base-font);
        font-size: $typographic-base-font-size;
        font-weight: normal;

        &:hover,
        &:focus {
          border-bottom: 1px solid $color-blue;
          color: $color-blue;
        }

        &.active {
          border-bottom: 1px solid var(--color-typographic-base-font);
          color: var(--color-typographic-base-font);
        }
      }
    }
  }
}
