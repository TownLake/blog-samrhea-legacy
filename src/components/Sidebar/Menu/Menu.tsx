import React from "react";

import { Link } from "gatsby";

import * as styles from "./Menu.module.scss";

type Props = {
  menu: Array<{
    label: string;
    path: string;
  }>;
};

const Menu: React.FC<Props> = ({ menu }: Props) => (
  <nav className={styles.menu}>
    <ul className={styles.list}>
      {menu.map((item) => (
        <li className={styles.item} key={item.path}>
          <Link
            to={item.path}
            className={styles.link}
            activeClassName={styles.active}
          >
            {item.label}
          </Link>
        </li>
      ))}
    </ul>
  </nav>
);

export default Menu;
