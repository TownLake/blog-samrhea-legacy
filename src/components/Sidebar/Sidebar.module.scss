@import "../../assets/scss/mixins";

.sidebar {
  width: 100%;

  .inner {
    @include padding(1, 0.75, 0);
    position: relative;

    &:after {
      background: var(--color-sidebar-border);
      background: linear-gradient(
          to bottom,
          var(--color-sidebar-border) 0%,
          var(--color-sidebar-border) 48%,
          var(--color-sidebar-border-fade) 100%
      );
    }
  }
}

@include breakpoint-sm {
  .sidebar {
    lost-column: 5/12;

    .inner {
      @include padding(1.25, 0.75, 0);

      &:after {
        bottom: 0;
        content: "";
        height: 540px;
        position: absolute;
        right: -10px;
        top: 30px;
        width: 1px;
      }
    }
  }
}

@include breakpoint-md {
  .sidebar {
    lost-column: 1/3;

    .inner {
      @include padding-equal(1.5);
    }
  }
}
