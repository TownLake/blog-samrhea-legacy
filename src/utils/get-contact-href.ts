import { btoa } from "abab";

const getContactHref = (name: string, contact: string) => {
  const hrefs: { [key: string]: string } = {
    email: btoa(contact) || "",
    line: `line://ti/p/${contact}`,
    telegram: `https://t.me/${contact}`,
    vkontakte: `https://vk.com/${contact}`,
    medium: `https://medium.com/${contact}`,
    github: `https://github.com/${contact}`,
    weibo: `https://www.weibo.com/${contact}`,
    gitlab: `https://www.gitlab.com/${contact}`,
    codepen: `https://www.codepen.io/${contact}`,
    twitter: `https://www.twitter.com/${contact}`,
    soundcloud: `https://soundcloud.com/${contact}`,
    facebook: `https://www.facebook.com/${contact}`,
    instagram: `https://www.instagram.com/${contact}`,
    linkedin: `https://www.linkedin.com/in/${contact}`,
    youtube: `https://www.youtube.com/channel/${contact}`,
    mastodon: `${contact}`,
  };

  return hrefs[name] ?? contact;
};

export default getContactHref;
