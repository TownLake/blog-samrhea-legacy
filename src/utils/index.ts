export { default as getIcon } from "./get-icon";
export { default as getContactHref } from "./get-contact-href";
export { default as toKebabCase } from "./to-kebab-case";
export { default as getDefaultColorMode } from "./get-default-color-mode";
export * as testUtils from "./test-utils";
