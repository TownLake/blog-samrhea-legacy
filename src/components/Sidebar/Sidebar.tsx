import React from "react";

import { useSiteMetadata } from "@/hooks";

import { Author } from "./Author";
import { Contacts } from "./Contacts";
import { Copyright } from "./Copyright";
import { Menu } from "./Menu";

import * as styles from "./Sidebar.module.scss";

type Props = {
  isIndex?: boolean;
};

const Sidebar = ({ isIndex }: Props) => {
  const { author, copyright, menu } = useSiteMetadata();

  return (
    <div className={styles.sidebar}>
      <div className={styles.inner}>
        <Author author={author} isIndex={isIndex} />
        <Menu menu={menu} />
        <Contacts contacts={author.contacts} />
        <Copyright copyright={copyright} />
      </div>
    </div>
  );
};

export default Sidebar;
