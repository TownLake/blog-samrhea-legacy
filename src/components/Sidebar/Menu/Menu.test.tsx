import React from "react";

import { Menu } from "@/components/Sidebar/Menu";
import * as mocks from "@/mocks";
import { testUtils } from "@/utils";

describe("Menu", () => {
  test("renders correctly", () => {
    const props = { menu: mocks.menu };
    const tree = testUtils
      .createSnapshotsRenderer(<Menu {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
