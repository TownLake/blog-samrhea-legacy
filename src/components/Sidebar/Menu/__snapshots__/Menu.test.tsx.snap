// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu renders correctly 1`] = `
<nav>
  <ul>
    <li>
      <a
        href="/"
      >
        Articles
      </a>
    </li>
    <li>
      <a
        href="/pages/about"
      >
        About Me
      </a>
    </li>
    <li>
      <a
        href="/pages/contacts"
      >
        Contact Me
      </a>
    </li>
  </ul>
</nav>
`;
